File: webApp/src/app/common/ticket.ts
/**
 * 车票实体
 */
import { Route } from './route';
import { User } from './user';
import { Order } from './order';

export class Ticket {
  /** id */
  id: number;

  /** 车次编号 */
  route: Route;

  /** 用户编号 */
  user: User;

  /** 订单编号 */
  order: Order;

  /** 单价 */
  price: number;

  /** 车票状态
   * 0 已支付
   * 1 已退票
   * 2 以完成（不能退票）
   */
  status: number;

  constructor(data?: { id?: number, route?: Route, user?: User, order?: Order, price?: number, status?: number }) {
    if (data) {
      this.id = data.id;
      this.route = data.route;
      this.user = data.user;
      this.order = data.order;
      this.price = data.price;
      this.status = data.status;
    }
  }
}

File: bus-ticket/src/main/java/xiaoqiangZzz/busticket/Entity/Ticket.java
package xiaoqiangZzz.busticket.Entity;


import javax.persistence.*;

@Entity
public class Ticket {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(nullable = false)
    private Route route;

//    @ManyToOne
//    @JoinColumn(nullable = false)
//    private Orders orders;

    @ManyToOne
    @JoinColumn(nullable = false)
    private User user;

    private float price;

    private int status;


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Route getRoute() {
        return route;
    }

    public void setRoute(Route route) {
        this.route = route;
    }

//    public Orders getOrders() {
//        return orders;
//    }
//
//    public void setOrders(Orders orders) {
//        this.orders = orders;
//    }

    public float getPrice() {
        return price;
    }

    public void setPrice(float price) {
        this.price = price;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}
